table.cpp

```diff
@@ -12,10 +12,10 @@ using std::string;
 
 table_t::table_t(const string& conninfo, const string& name, const string& type, const columns_t& columns, const hstores_t& hstore_columns,
     const int srid, const int scale, const bool append, const bool slim, const bool drop_temp, const int hstore_mode,
-    const bool enable_hstore_index, const boost::optional<string>& table_space, const boost::optional<string>& table_space_index) :
+    const bool enable_hstore_index, const boost::optional<string>& table_space, const boost::optional<string>& table_space_index, const boost::optional<string>& schema) :
```

Comment:
I had a look, moving to `table_options_t` would get `hstore_columns`, `hstore_mode`, `enable_hstore_index`, `tblsmain_data`, and `tblsmain_index` into one struct for sure.

`name`, `type`, `columns`, and `SRID` aren't part of `options_t` but might make sense in `table_options_t`, but would definitely differ between different tables, even in the pgsql backend.

`conninfo` could be per table (and probably should, as it opens up interesting options), but might require more work to move.

`append`, `slim` and `droptemp` are needed by the table constructor, but have more impact on the *middle* tables, not the output ones.